setup/context/commands.go

```diff
@@ -408,29 +402,55 @@ func Endpoint(c *Context, syntax string, insecure, dev, quiet bool) error {
 }
 
 func Plan(c *Context, args []string) error {
-	if err := checkDockercfg(c, false); err != nil {
+	if err := checkDockercfg(c, c.InstanceDir, false); err != nil {
 		return err
 	}
 
 	args = append([]string{"plan"}, args...)
 	return Terraform(c, args)
 }
 
-func checkDockercfg(c *Context, force bool) error {
-	path := fmt.Sprintf("%s/dockercfg", c.InstanceDir)
+// checkDockercfg looks for docker config file (config.json or dockercfg) from
+// given directory and saves in `dockercfg` format in instance directory.
+func checkDockercfg(c *Context, dockerCfgDir string, force bool) error {
+	if dockerCfgDir == "" {
```

Comment:
I have seen this pattern in go source code, and I was modeling from it. Definitely will change if you prefer it differently, as I want to set this up the proper way. Thanks for the tip.

Moving default logic to caller.